### PR TITLE
finished the vizualization library

### PR DESCRIPTION
**Pull request by ymaheshwari2007**
## Changes

_Explanation of changes goes here_
Added a data loading library/file for now and refactor the code for everything to fit the library and remove unnecessary lines of code.

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## Screenshots

_If you made UI changes you must post a screenshot of the whole page for each change in 1) a normal sized window and 2) the smallest possible window_

_If you did any manual testing (e.g., with Postman), put screenshots of the http request and before and after of the db_

_If none of this applies, you can delete this section_

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ x] All commits are tagged with the ticket number
- [ x] No linting errors / newline at end of file warnings
- [ x] All code follows repository-configured prettier formatting
- [ x] No merge conflicts
- [ x] All checks passing
- [ x] Screenshots of UI changes (see Screenshots section)
- [ x] Remove any non-applicable sections of this template
- [ x] Assign the PR to yourself
- [ x] No `package-lock.json` changes (unless dependencies have changed)
- [ x] Request reviewers & ping on Slack
- [ x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: analysis/src/telemetry_overlay/data_loader_class.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    load_dotenv()
+    link = os.getenv(key='DATABASE_URL')
+    try:
+        conn = psycopg2.connect(link)
+        print('connected to db')
+    except:
+        print('not connect')
+
+    @classmethod
+    def get_data(cls, sensor, table='data', start_time=None, end_time=None):
```
Comment (bracyw) [inline analysis/src/telemetry_overlay/data_loader_class.py]: can we name this sensor_topic (to be clear to everyone, firmware knows these as topics, and then engineers will understand the sensor part)

File: analysis/src/telemetry_overlay/data_loader_class.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            start_time (str, optional): Start of time range (e.g. '2025-06-08 15:17:09.00').
+                Required when table='data'.
+            end_time (str, optional): End of time range.
+                Required when table='data'.
+
+        Returns:
+            pd.DataFrame: Raw query results as a DataFrame.
+        """
+        if table == 'data_type':
+            q = f"SELECT * FROM data_type WHERE data_type.\"name\" = '{sensor}'"
```
Comment (bracyw) [inline analysis/src/telemetry_overlay/data_loader_class.py]: can you convert this to pscop2 cursor execute command with params to avoid any sql injection. Probably shouldn't be a problem but could happen as an accident.

File: analysis/src/telemetry_overlay/data_loader_class.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            return to_return
+        return wrapper
+
+    # Database connection — established once at class load time.
+    load_dotenv()
+    link = os.getenv(key='DATABASE_URL')
+    try:
+        conn = psycopg2.connect(link)
+        print('connected to db')
+    except:
```
Comment (bracyw) [inline analysis/src/telemetry_overlay/data_loader_class.py]: after printing connection fail, can you raise the error again to avoid silent failure.

**Review by bracyw — CHANGES_REQUESTED** [review]
Looks good just make

**Review by bracyw — APPROVED** [review]
LGTM 🔥
